pkg/daemon/controller/remote.go

```diff
@@ -0,0 +1,177 @@
+package controller
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+
+	ramav1 "github.com/oecp/rama/pkg/apis/networking/v1"
+	"github.com/oecp/rama/pkg/daemon/containernetwork"
+	"github.com/vishvananda/netlink"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog"
+)
+
+type RemoteVtepCache struct {
+	remoteVtepIPMap map[string]net.HardwareAddr
+
+	ch chan struct{}
+}
+
+func NewRemoteVtepCache() *RemoteVtepCache {
+	return &RemoteVtepCache{
+		remoteVtepIPMap: map[string]net.HardwareAddr{},
+		ch:              make(chan struct{}, 1),
+	}
+}
+
+func (rvc *RemoteVtepCache) UpdateRemoteVtepIPs(remoteVteps []*ramav1.RemoteVtep) error {
+	rvc.lock()
+	defer rvc.unlock()
```

Comment:
Don't lock in such a big range, this will slow the speed. Use lock in a smaller range. Like extraing a function called "setRemoteVtep", and lock in that function.

Reply:
why? i can't understand, pls explain more

Reply:
Increasing the capacity of `map` involves rehasing.